src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerActionCounts.cs

```diff
@@ -174,6 +181,22 @@ internal class AnalyzerActionCounts
         /// </summary>
         public bool HasAnyExecutableCodeActions { get; }
 
+        /// <summary>
+        /// Returns true if there are any analyzer action callbacks that are driven by compilation events,
+        /// such as <see cref="SymbolDeclaredCompilationEvent"/>, <see cref="CompilationUnitCompletedEvent"/>, etc.
+        /// Many callbacks into the diagnostics analyzers are driven in the <see cref="AnalyzerDriver"/>
+        /// by compilation events added to the <see cref="Compilation.EventQueue"/>. For these callbacks to be executed,
+        /// the analyzer driver host needs to force complete the events in the relevant part of the compilation,
+        /// i.e. relevant tree(s) or entire compilation. This force complete operation incurs a performance cost,
+        /// which can be avoided if the analyzer(s) to be executed do not register any actions which are driven by
+        /// compilation events.
```

Comment:
Consider including an example here. Perhaps "if the analyzer(s) to be executed, such as syntax-only analyzers, do not register any actions which are driven by compilation events."